include/hipSYCL/runtime/settings.hpp

```diff
@@ -28,29 +28,24 @@
 #ifndef HIPSYCL_RT_SETTINGS_HPP
 #define HIPSYCL_RT_SETTINGS_HPP
 
+#include "hipSYCL/runtime/device_id.hpp"
 #include <ios>
 #include <optional>
 #include <string>
 #include <cstdlib>
 #include <algorithm>
 #include <sstream>
 #include <iostream>
+#include <vector>
 namespace hipsycl {
 namespace rt {
 
 enum class scheduler_type { direct };
 
-inline std::istream &operator>>(std::istream &istr, scheduler_type &out) {
-  std::string str;
-  istr >> str;
-  if (str == "direct")
-    out = scheduler_type::direct;
-  else
-    istr.setstate(std::ios_base::failbit);
-  return istr;
-}
+std::istream &operator>>(std::istream &istr, scheduler_type &out);
+std::istream &operator>>(std::istream &istr, std::vector<rt::backend_id> &out);
 
-enum class setting { debug_level, scheduler_type };
+enum class setting { debug_level, scheduler_type, active_backends };
```

Comment:
Yeah, makes sense to use something more standard. I initially thought about the extension path would be to go to a `--hipsycl-targets` compatible syntax for device selection, except that it uses device ids instead of architectures of course, like `cuda:0,1;hip:4` or similar. But for now, thought it'd be nice to just be able to select backends (and devices could be _easily_ selected via the known backend variables, but then again, you could use those to basically disable the backend as well :D).

As of [DPC++ docs](https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md) `SYCL_DEVICE_FILTER` uses `backend:device_type:device_num,backend:device_type:device_num,..` as syntax, which feels slightly overkill for hipSYCL as we currently only have one device_type per backend.. but OpenMP offload and HIP for FPGAs might become a thing, right?
What annoys me slightly with this is, that you'd have to specify the full triple multiple times to select multiple devices of the same backend, e.g. `cuda:gpu:0,cuda:gpu:1` and that they use `,` for triple splitting which is unintuitive given the `hipsycl-targets` syntax.. 

So we could of course support `SYCL_DEVICE_FILTER` for compatibility.. but have `HIPSYCL_VISIBLE_DEVICES` use the `hipsycl-targets` syntax. What are your thoughts, which one we should actually support?
And I guess I should probably try to revamp the PR to include preliminary support for device selection as well.

Reply:
Decided to use `HIPSYCL_VISIBILITY_MASK` with `hipsycl-targets` like syntax for now, with the expectation to implement compatibility options, when we can properly support device selection.